iothub/device/src/DirectMethod/EdgeModuleDirectMethodRequest.cs

```diff
@@ -23,25 +26,76 @@ public EdgeModuleDirectMethodRequest(string methodName)
         /// a direct method on an edge device or an edge module connected to the same edge hub.
         /// </summary>
         /// <param name="methodName">The method name to invoke.</param>
-        /// <param name="payload">The direct method payload bytes.</param>
-        public EdgeModuleDirectMethodRequest(string methodName, byte[] payload)
-            : base(methodName)
+        /// <param name="payload">The direct method payload that will be serialized using <see cref="DefaultPayloadConvention"/>.</param>
+        public EdgeModuleDirectMethodRequest(string methodName, object payload)
         {
+            MethodName = methodName;
             Payload = payload;
         }
 
         /// <summary>
-        /// A direct method request to be initialized by the client application when using an <see cref="IotHubModuleClient"/> for invoking
-        /// a direct method on an edge device or an edge module connected to the same edge hub.
+        /// The method name to invoke.
         /// </summary>
-        /// <param name="methodName">The method name to invoke.</param>
-        /// <param name="payload">The direct method payload that will be serialized and encoded per <see cref="IotHubClientOptions.PayloadConvention"/>.</param>
-        public EdgeModuleDirectMethodRequest(string methodName, object payload)
-            : base(methodName)
-        {
-            Payload = payload == null
-                ? null
-                : PayloadConvention.GetObjectBytes(payload);
-        }
+        [JsonProperty("methodName")]
+        public string MethodName { get; private set; }
```

Comment:
why private set and not readonly?

Reply:
Newtonsoft does persist readonly properties during serializtion by default; however, someone might unintentionally override that behavior by setting "JsonSerializerOptions.IgnoreReadOnlyProperties = true" on the serializer options set by their application (they might want to do that for their custom json object behavior).
That would be the only reason why I'd want us to leave this private setter as-is.

What do you think? @drwill-ms

Reply:
I'm not sure we employ that logic with other models, but I see your point. I'd simply suggest we be consistent across our models.

Reply:
You're right, our current approach has been to use readonly properties. I can update this to match the other models and we decide to change our approach if we run into an issue in the future. (It is a private set, so it wouldn't be a breaking change).